syntax = "proto3";

package user;

service UserService {
  rpc GetUser (GetUserRequest) returns (User);
  rpc CreateUser (CreateUserRequest) returns (User);
  rpc GetUsers (GetUsersRequest) returns (GetUsersResponse);
}

message User {
  int32 id = 1;
  string name = 2;
  string email = 3;
  bool is_active = 4;
}

message GetUserRequest {
  int32 id = 1;
}

message CreateUserRequest {
  string name = 1;
  string email = 2;
}

message GetUsersRequest {
  int32 limit = 1;
  int32 offset = 2;
}

message GetUsersResponse {
  repeated User users = 1;
}
